dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/file/ImmediateFileWriterTest.kt

```diff
@@ -64,97 +61,62 @@ internal class FileWriterTest {
-        underTest.invokeMethod(
-            "consumeQueue",
-            methodEnclosingClass = LazyHandlerThread::class.java
-        ) // force the lazy handler thread to consume all the queued messages
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        rootDir.deleteRecursively()
     }
 
     @Test
-    fun `migrates the data before doing anything else`(forge: Forge) {
-        val modelValue = forge.anAlphabeticalString()
+    @TestTargetApi(Build.VERSION_CODES.O)
+    fun `writes a valid model`(forge: Forge) {
+        val model = forge.anAlphabeticalString()
         val fileNameToWriteTo = forge.anAlphaNumericalString()
-        val file = generateFile(fileNameToWriteTo)
+        val file = File(rootDir, fileNameToWriteTo)
         whenever(mockedOrchestrator.getWritableFile(any())).thenReturn(file)
 
-        // when
-        underTest.write(modelValue)
+        underTest.write(model)
 
-        // then
-        val inOrder = inOrder(mockDataMigrator, mockedSerializer)
-        inOrder.verify(mockDataMigrator).migrateData()
-        inOrder.verify(mockedSerializer).serialize(modelValue)
+        assertThat(file.readText())
+            .isEqualTo(model)
     }
 
     @Test
-    fun `writes a valid mode`(forge: Forge) {
-        val modelValue = forge.anAlphabeticalString()
+    @TestTargetApi(Build.VERSION_CODES.O)
+    fun `writes several models`(forge: Forge) {
+        val models = forge.aList { anAlphabeticalString() }
         val fileNameToWriteTo = forge.anAlphaNumericalString()
-        val file = generateFile(fileNameToWriteTo)
+        val file = File(rootDir, fileNameToWriteTo)
         whenever(mockedOrchestrator.getWritableFile(any())).thenReturn(file)
 
-        // when
-        underTest.write(modelValue)
+        models.forEach {
+            underTest.write(it)
+        }
 
-        // then
-        // TODO: RUMM-135 assert also the content of the file. We will not do it now because
-        //  we will remove the obfuscation anyway.
-        assertThat(file.readText()).isNotEmpty()
+        assertThat(file.readText())
+            .isEqualTo(models.joinToString(","))
     }
 
     @Test
+    @TestTargetApi(Build.VERSION_CODES.O)
     fun `does nothing when SecurityException was thrown while providing a file`(
         forge: Forge,
         @SystemOutStream outputStream: ByteArrayOutputStream
     ) {
         val modelValue = forge.anAlphabeticalString()
-        val securityException = SecurityException(forge.anAlphabeticalString())
-
-        doThrow(securityException).whenever(mockedOrchestrator).getWritableFile(any())
-
-        // when
-        underTest.write(modelValue)
-
-        // then
-        if (BuildConfig.DEBUG) {
-            val logMessages = outputStream.toString().trim().split("\n")
-            assertThat(logMessages[0]).matches("E/DD_LOG: FileWriter: Couldn't access file.*")
-        }
-    }
-
-    @Test
-    fun `does nothing when FileOrchestrator returns a null file`(
```

Comment:
Have you deleted this ? Or is just misleading on Github ?

Reply:
Yes it was deleted, but I'll add it back in the ImmediateFileWriter